### ISSUE TITLE
Cloudflare worker Uncaught ReferenceError: FinalizationRegistry is not defined

### ISSUE THREAD

**Issue by EiffelFly** [dqbd/tiktoken#46]
Hi, thanks for building this tool! I had encountered a problem when I tried to run this repo in cloudflare worker. 

you could go to this repo to reproduce [repo](https://github.com/EiffelFly/mark-your-frog)

- `pnpm i` at the root folder
- go into /packages/api folder and `pnpm dev`
- select `turn on local mode` in the terminal

Then you will see the issue happening. 

It is quite hard to find the solution on the web, so I am curious about how do you test the package on Cloudflare worker and make it work.

**Comment by KittenYang** [comment]
same here

**Comment by KittenYang** [comment]
back to this repo: https://github.com/niieani/gpt-tokenizer

**Comment by dqbd** [comment]
Hi! This does seem to be a regression caused by 1.0.7, will address this issue in later versions. 

In the meantime, [`js-tiktoken`](https://www.npmjs.com/package/js-tiktoken) is a pure JS version of the package, which should be more suitable for edge runtimes such as Cloudflare Workers.

**Comment by EiffelFly** [comment]
@dqbd I can confirm js-toktoken is working correctly in edge-runtime due to langchainjs use this package under the hood!! Thanks for the head up.

**Comment by iwasrobbed** [comment]
Thanks @dqbd ! I was a bit confused as well since the readme still points towards using the wasm version

Is the plan to maintain both wasm and JS going forward?

**Comment by dqbd** [comment]
Yep! Both of the libraries have their advantages and disadvantages and the plan is to support both for the foreseeable future.

**Comment by gablabelle** [comment]
@dqbd I'm having the same issue on Vercel (edge function). Since you've closed the issue, what was the resolution of the problem?

**Comment by poweroutlet2** [comment]
Still running into this error using Vercel edge functions. Any updates here? Using js-tiktoken seems to make the edge function size >1MB
